tests/unit/cluster/slot-ownership.tcl

```diff
@@ -59,3 +59,40 @@ start_cluster 2 2 {tags {external:skip cluster}} {
         }
     }
 }
+
+start_cluster 3 1 {tags {external:skip cluster} overrides {shutdown-timeout 100}} {
+    test "Primary will not crash when processing dirty slots during shutdown waiting" {
+        R 0 set FOO 0
+
+        # Pause the replica.
+        pause_process [srv -3 pid]
+
+        # Incr the key and immediately shutdown the primary.
```

Comment:
Some explanation to help the readers if they are not familiar with the shutdown procedure.

```suggestion
        # Incr the key and immediately shutdown the primary.
        # The primary waits for the replica to replicate before exiting.
```